### PR TITLE
New command line commands and options

### PR DESCRIPTION
**Pull request by p-leroy**
new option SPLIT_XY_Z for commands C2C_DIST and C2M_DIST
new command SF_OP_SF
new command SF_INTERP with option DEST_IS_FIRST
new command SF_ADD_CONST

### CODE REVIEW

File: qCC/ccComparisonDlg.cpp [hunk 1]
```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
 				}
 			}
 			ccLog::Warning("[ComputeDistances] Result has been split along each dimension (check the 3 other scalar fields with '_X', '_Y' and '_Z' suffix!)");
+            if (mergeXY)
+            {
+                ccLog::Warning("[ComputeDistances] compute 2D distance (xy plane)");
+                int sf2D = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (XY)")));
+                if (sf2D >= 0)
```
Comment (dgirardeau) [inline qCC/ccComparisonDlg.cpp:954]: if (sf2D < 0)
         sf2D = m_compCloud->addScalarField(qPrintable(m_sfName + QString(" (XY)")));

(no need to delete the existing one?)

File: qCC/ccComparisonDlg.cpp [hunk 2]
```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
...
+                    m_compCloud->deleteScalarField(sf2D);
+                sf2D = m_compCloud->addScalarField(qPrintable(m_sfName + QString(" (XY)")));
+                CCCoreLib::ScalarField* sf = m_compCloud->getScalarField(sf2D);
+                for (int idx = 0; idx < m_compCloud->size(); idx++)
```
Comment (dgirardeau) [inline qCC/ccComparisonDlg.cpp:958]: theoretically you should check that 'sf' exists (just in case)

File: qCC/ccComparisonDlg.cpp [hunk 3]
```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
...
+                {
+                    float d2D = pow(pow(c2cParams.splitDistances[0]->getValue(idx), 2) +  pow(c2cParams.splitDistances[1]->getValue(idx), 2), 0.5);
+                    sf->setValue(idx, d2D);
+                }
+                sf->computeMinAndMax();
+                // remove X and Y SF
+                int sfX = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (X)")));
+                if (sfX >= 0)
+                    m_compCloud->deleteScalarField(sfX);
```
Comment (dgirardeau) [inline qCC/ccComparisonDlg.cpp:967]: same comment as above

File: qCC/ccComparisonDlg.cpp [hunk 4]
```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
...
+                int sfY = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (Y)")));
+                if (sfY >= 0)
+                    m_compCloud->deleteScalarField(sfY);
```
Comment (dgirardeau) [inline qCC/ccComparisonDlg.cpp:970]: same

File: qCC/ccEntityAction.cpp [hunk 5]
```diff
@@ -516,6 +516,59 @@ namespace ccEntityAction
 		return true;
 	}
 	
+    //! Interpolate scalar fields from on entity and transfer them to another one without the dialog
```
Comment (dgirardeau) [inline qCC/ccEntityAction.cpp:519]: from on**e**

File: qCC/ccEntityAction.cpp [hunk 6]
```diff
@@ -516,6 +516,59 @@ namespace ccEntityAction
...
+    bool	interpolateSFs(ccPointCloud *source, ccPointCloud *dest, int sfIndex, QWidget* parent)
+    {
+        if (!source || !dest)
+        {
+            ccConsole::Error(QObject::tr("Unexpected empty cloud pointers!"));
```
Comment (dgirardeau) [inline qCC/ccEntityAction.cpp:524]: empty cloud pointers --> null cloud pointers

File: qCC/ccEntityAction.cpp [hunk 7]
```diff
@@ -516,6 +516,59 @@ namespace ccEntityAction
...
+            return false;
+        }
+
+        if (!source->hasScalarFields())
+        {
+            ccConsole::Error(QObject::tr("[ccEntityAction::interpolateSFs] The source cloud has no scalar field!"));
+            return false;
+        }
+
+        std::vector<int> sfIndexes;
```
Comment (dgirardeau) [inline qCC/ccEntityAction.cpp:534]: std::vector<int> sfIndexes(1);

File: qCC/ccEntityAction.cpp [hunk 8]
```diff
@@ -516,6 +516,59 @@ namespace ccEntityAction
...
+
+        unsigned sfCount = source->getNumberOfScalarFields();
+        if (sfIndex > sfCount)
+        {
+            ccConsole::Error(QObject::tr("[ccEntityAction::interpolateSFs] Invalid scalar field index!"));
+            return false;
+        }
+        else
+            sfIndexes.push_back(sfIndex);
```
Comment (dgirardeau) [inline qCC/ccEntityAction.cpp:543]: sfIndexes.back() = sfIndex;

Comment (dgirardeau) [reply qCC/ccEntityAction.cpp:543]: and you can remove the else above

File: qCC/ccEntityAction.cpp (hunk 7 above)
Comment (dgirardeau) [reply qCC/ccEntityAction.cpp:534]: and move it just before line 543

File: qCC/ccEntityAction.cpp (hunk 5 above)
Comment (dgirardeau) [reply qCC/ccEntityAction.cpp:519]: and don't you have a dialog still?

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: libs/qCC_io/cmake/GDALSupport.cmake [hunk 9]
```diff
@@ -31,7 +31,7 @@ function( target_link_GDAL ) # ARGV0 = project name
 		file( GLOB GDAL_DLL_FILES ${GDAL_BIN_DIR}/gdal*.dll )
 message( ${GDAL_DLL_FILES} )
                 message( STATUS "GDAL VERSION: " ${GDAL_VERSION} )
-                if (GDAL_VERSION EQUAL 3.2)
+                if (GDAL_VERSION EQUAL 3.4)
```
[the remark this answers is not in the document]
Comment (p-leroy) [reply libs/qCC_io/cmake/GDALSupport.cmake:34]: I added my 3.4 block.

File: qCC/ccComparisonDlg.cpp (hunk 2 above)
Comment (p-leroy) [reply qCC/ccComparisonDlg.cpp:958]: exact

File: qCC/ccComparisonDlg.cpp (hunk 4 above)
Comment (p-leroy) [reply qCC/ccComparisonDlg.cpp:970]: I wanted to remove y because the user wants xy distances, maybe it is use less to keep it?

Comment (dgirardeau) [reply qCC/ccComparisonDlg.cpp:970]: By principle, I wouldn't remove an existing scalar field silently (and the user can delete it manually).

File: qCC/ccComparisonDlg.cpp (hunk 3 above)
Comment (dgirardeau) [reply qCC/ccComparisonDlg.cpp:967]: By principle, I wouldn't remove an existing scalar field silently (and the user can delete it manually).

Comment (p-leroy) [reply qCC/ccComparisonDlg.cpp:967]: OK

File: qCC/ccComparisonDlg.cpp (hunk 4 above)
Comment (p-leroy) [reply qCC/ccComparisonDlg.cpp:970]: Oki doki

File: qCC/ccEntityAction.cpp (hunk 5 above)
Comment (p-leroy) [reply qCC/ccEntityAction.cpp:519]: You have a dialog but it is set automatically and it does not block the command line.

File: qCC/ccEntityAction.cpp (hunk 6 above)
Comment (p-leroy) [reply qCC/ccEntityAction.cpp:524]: ok

File: qCC/ccEntityAction.cpp (hunk 8 above)
Comment (p-leroy) [reply qCC/ccEntityAction.cpp:543]: nice!

File: qCC/ccEntityAction.cpp (hunk 7 above)
Comment (p-leroy) [reply qCC/ccEntityAction.cpp:534]: done

**Review by dgirardeau — APPROVED** [review]

**Comment by dgirardeau** [comment]
Thanks for the contribution

**Pull request merged**